cmd/gitserver/server/gitservice.go

```diff
@@ -0,0 +1,129 @@
+package server
+
+import (
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/inconshreveable/log15"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var uploadPackArgs = []string{
+	// Partial clones/fetches
+	"-c", "uploadpack.allowFilter=true",
+
+	// Can fetch any object. Used in case of race between a resolve ref and a
+	// fetch of a commit. Safe to do, since this is only used internally.
+	"-c", "uploadpack.allowAnySHA1InWant=true",
+
+	"upload-pack",
+
+	"--stateless-rpc", "--strict",
+}
+
+// handleGitService is a smart Git HTTP transfer protocol as documented at
```

Comment:
I think this name is outdated?

```suggestion
// gitServiceHandler is a smart Git HTTP transfer protocol handler as documented at
```